src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed partial class ConvertSwitchStatementToExpressionDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public ConvertSwitchStatementToExpressionDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.ConvertSwitchStatementToExpressionDiagnosticId,
+                new LocalizableResourceString(nameof(CSharpFeaturesResources.Convert_switch_statement_to_expression), CSharpFeaturesResources.ResourceManager, typeof(CSharpFeaturesResources)),
+                new LocalizableResourceString(nameof(CSharpFeaturesResources.Use_switch_expression), CSharpFeaturesResources.ResourceManager, typeof(CSharpFeaturesResources)))
+        {
+        }
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterOperationAction(AnalyzeOperation, OperationKind.Switch);
+
+        private void AnalyzeOperation(OperationAnalysisContext context)
+        {
+            var switchOperation = (ISwitchOperation)context.Operation;
+            if (switchOperation.Syntax.ContainsDiagnostics)
+            {
+                return;
+            }
+
+            var analysisResult = Analyzer.Analyze(switchOperation);
+            if (analysisResult.IsFailure || analysisResult.IsNeutral)
+            {
+                return;
+            }
+
+            context.ReportDiagnostic(Diagnostic.Create(Descriptor,
+                switchOperation.Syntax.GetFirstToken().GetLocation()));
```

Comment:
oh, i see.  this is how you're getting the node.  not a huge fan.  would prefer to use .AdditionalLocations for this.

Reply:
what to pass when no additional locations are required? this is straight from another analyzer, addbraces, I think.

Reply:
it's not that they're not required, it's just that this feels cleaner to me in that both sides totally agree on what's going in and what's going out.  THe analyzer says "AdditionalLocations" points at the SwitchStatement, and the fixre has the right span to get teh switch.  i'm not a fan of going "token => node" in the same way as you're doing here.

Reply:
would still like this :)